### PR TITLE
Create Favorite Button and Screen

### PR DESCRIPTION
**Pull request by myin142**
The Favorite Screen loads every station that is listed in the AsyncStorage. I created a favorite button on StationList that changes in realtime. I kept the favorite button on Plan that currently does not change in realtime.

### CODE REVIEW

File: src/components/Plan.js [hunk 1]
```diff
@@ -54,6 +92,18 @@ class Plan extends Component {
       hackLocationButton: 0
     }),500);
   }
+
+  componentDidMount(){
+    AsyncStorage.getItem(FAV_KEY).then(res =>{
```
Comment (tomekz) [inline src/components/Plan.js]: It is a good practice to create an abstraction on top of 'AsyncStorage'. Maybe  you can try this as reference https://github.com/jasonmerino/react-native-simple-store/blob/master/src/index.js

File: src/components/Plan.js [hunk 2]
```diff
@@ -38,6 +41,41 @@ class Plan extends Component {
     }
   }
 
+  onFavPress(station){
```
Comment (tomekz) [inline src/components/Plan.js:44]: This logic to store a single station is duplicated in ‘Plan.js‘ and 'StationCard.js'.
Would be nice to extract it to the wrapper mentioned in other comment where we could have a method e.g. like
´addToFavourites (station)´

File: src/components/Plan.js [hunk 3]
```diff
@@ -73,12 +123,26 @@ class Plan extends Component {
...
               }}
               image = { this.getStationIcon(station) }
-              title={`${station.extra.internal_id} ${station.name}`}
-              description={`${station.free_bikes} bikes | ${station.empty_slots} empty slots`}
               key={station.id}
               ref={station.id}
               onPress={this.onMarkerPress.bind(this)}
               >
+
+              <MapView.Callout onPress={() => this.onFavPress(station)}>
```
Comment (tomekz) [inline src/components/Plan.js:112]: I like your idea of introducing Callout. However have you tested it on Android Virtual Device or even better on actual device. Is the map being rendered smoothly without any glitches?
Cause that's why I havn't used them initially

**Review by tomekz — CHANGES_REQUESTED** [review]

File: src/components/Plan.js (hunk 3 above)
Comment (myin142) [reply src/components/Plan.js:112]: I tested it on my old samsung galaxy s4 and it looks normal to me.

**Comment by myin142** [comment]
I have now moved the AsyncStorage to its own file. On my own android device the MapView.Callout looks normal so I kept it.

**Review by tomekz — APPROVED** [review]

**Comment by tomekz** [comment]
@myin142 Thanks for your contribution. Now I can plan the next release and hopefully will publish it to play store soon

**Pull request merged**